notes-manager/src/js/reducers/index.js

```diff
@@ -0,0 +1,37 @@
+import { ADD_NOTE } from '../constants/action-types';
+import { REMOVE_NOTE } from '../constants/action-types';
+
+const initalState = {
+    notes: [],
+};
+
+function findIndex(notes, note_to_remove) {
+    for(let index=0;index<notes.length;index++) {
+        if (notes[index].title === note_to_remove.title) {
+            return index;
+        }
+    }
+}
+
+function rootReducer(state=initalState, action) {
+    
+    if (action.type === ADD_NOTE) {
+        return Object.assign({}, state, {
```

Comment:
While Object.assign is a way to do this, a more easy to read way is to use the spread `...` operator.

So, in this case, {nodes: [...state.nodes, action.note]}